storm-core/src/jvm/org/apache/storm/utils/ConfigUtils.java

```diff
@@ -135,7 +137,28 @@ public static int samplingRate(Map conf) {
         throw new IllegalArgumentException("Illegal topology.stats.sample.rate in conf: " + rate);
     }
 
-    // public static mkStatsSampler // depends on Utils.evenSampler() TODO, this is sth we need to do after util
+    public static Callable<Boolean> evenSampler(final int samplingFreq) {
+        final int start = 0;
+        final Random random = new Random();
+        final MutableInt curr = new MutableInt(-1);
+        final MutableInt target = new MutableInt(random.nextInt(samplingFreq));
```

Comment:
The entire reason for the MutableInt was so clojure could ack more like java.  It is going to be a lot faster to just use int instead.

```
    public static Callable<Boolean> evenSampler(final int samplingFreq) {
        final Random random = new Random();
        return new Callable<Boolean>() {
            private int curr = -1;
            private int target = random.nextInt(samplingFreq);

            @Override
            public Boolean call() throws Exception {
                curr++;
                if (curr >= samplingFreq) {
                    curr = 0;
                    target = random.nextInt(samplingFreq);
                }
                return curr == target;
            }
        };
```